controlplane/internal/controllers/rke2controlplane_controller_test.go

```diff
@@ -251,4 +258,129 @@ var _ = Describe("Reconcile control plane conditions", func() {
 		Expect(conditions.GetMessage(rcp, controlplanev1.ControlPlaneComponentsHealthyCondition)).To(Equal(
 			"Control plane node missing-machine does not have a corresponding machine"))
 	})
+
+	It("should rotate kubeconfig secret if needed", func() {
+		r := &RKE2ControlPlaneReconciler{
+			Client:                    testEnv.GetClient(),
+			Scheme:                    testEnv.GetScheme(),
+			managementCluster:         &rke2.Management{Client: testEnv.GetClient(), SecretCachingClient: testEnv.GetClient()},
+			managementClusterUncached: &rke2.Management{Client: testEnv.GetClient()},
+		}
+		clusterName := client.ObjectKey{Namespace: ns.Name, Name: "test"}
+		endpoint := clusterv1.APIEndpoint{Host: "1.2.3.4", Port: 6443}
+
+		// Create kubeconfig secret with short expiry
+		shortExpiryDate := time.Now().Add(24 * time.Hour) // 1 day from now
+		secret, err := createKubeconfigSecret(ns.Name, shortExpiryDate)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(testEnv.Create(ctx, secret)).To(Succeed())
+
+		// Check that rotation is needed
+		needsRotation, err := kubeconfig.NeedsClientCertRotation(secret, certs.ClientCertificateRenewalDuration)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(needsRotation).To(BeTrue())
+
+		// Rotate kubeconfig secret
+		_, err = r.reconcileKubeconfig(ctx, clusterName, endpoint, rcp)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(testEnv.Get(ctx, client.ObjectKey{Namespace: ns.Name, Name: secret.Name}, secret)).To(Succeed())
```

Comment:
Can we also check that the kubeconfig secret is actually rotated? It seems to me that the current validation equals to “no rotation” test case: https://github.com/rancher/cluster-api-provider-rke2/pull/550/files#diff-cd4559684c03b3b2a238ed4f8c142bb867312c79174f5a6474b157ac7849a737R312-R316

Reply:
`reconcileKubeconfig` calls  `kubeconfig.CreateSecretWithOwner`, I'm also not sure that cert would get rotated.